drivers/memc/memc_mcux_flexspi.c

```diff
@@ -143,14 +143,11 @@ static int memc_flexspi_init(const struct device *dev)
 		return 0;
 	}
 
-	/* Apply pinctrl state */
 #ifdef CONFIG_PINCTRL
-	int ret;
-
-	ret = pinctrl_apply_state(data->pincfg, PINCTRL_STATE_DEFAULT);
-	if (ret < 0) {
-		return ret;
-	}
+	/* Some SOCs use internal flash, and don't define pin settings,
+	 * so don't check return code.
+	 */
+	pinctrl_apply_state(data->pincfg, PINCTRL_STATE_DEFAULT);
```

Comment:
```suggestion
	(void)pinctrl_apply_state(data->pincfg, PINCTRL_STATE_DEFAULT);
```

still, I think you should make the `pinctrl_apply_state` code conditional to SoCs supporting it, so that you get error handling on them, e.g.

```c
#if defined(CONFIG_PINCTRL) && (defined(MY_SOC_X) || defined(MY_SOC_Y))
/* do pinctrl */
#endif
```
(or reverse logic)

Reply:
This approach is fine with me. I left the return code check in place, and now only skip pin control states for the RT1024 and RT1064 SOCs

Reply:
@gmarull @danieldegrasse this actually breaks the case where boards have multiple `flexspi` controllers enabled.